backend/src/App.hs

```diff
@@ -0,0 +1,21 @@
+module App where
+
+import Control.Monad.Logger (LoggingT (runLoggingT), filterLogger)
+import Database.Esqueleto.Experimental (SqlPersistT)
+import Database.Esqueleto.Experimental qualified as SQL
+import Database.Redis qualified as Redis
+import Logging (LogLevelFilter, Logger, filterInfo)
+import Servant (Handler)
+
+data Env = Env {dbPool :: SQL.ConnectionPool, redisPool :: Redis.Connection, logger :: Logger}
+
+type AppM = ReaderT Env (LoggingT Handler)
+
+runAppLogger :: (MonadIO m) => LogLevelFilter -> Logger -> LoggingT m a -> m a
```

Comment:
A function like this (the critical part being the `LoggingT m a -> m a` bit) is necessary for "unwrapping" the logging part of our monad stack. Just like how `runReaderT` unwraps the Reader monad. You can see how these compose together in the `nt` function below. And you can also see how we use it to shim our logger into our (now slightly taller) SQL monad stack for runDb.

Reply:
✨